chainntnfs/txconfnotifier.go

```diff
@@ -297,6 +322,42 @@ func (tcn *TxConfNotifier) ConnectTip(blockHash *chainhash.Hash,
 		}
 	}
 
+	// In order to update the height hint for all the required transactions
+	// under one DB transaction, we'll gather the set of unconfirmed
+	// transactions along with the ones that confirmed at the current
+	// height.
+	var txs []chainhash.Hash
+	for confirmedTx := range tcn.txsByInitialHeight[tcn.currentHeight] {
+		txs = append(txs, confirmedTx)
+	}
+out:
+	for maybeUnconfirmedTx := range tcn.confNotifications {
```

Comment:
it is a bit hard to parse what is going on here..

Reply:
Hmm, would say that it's still a bit hard to parse the first time around.

Reply:
Generally, it seems like you can just update the _entire_ set of outstanding notifications at this instance. The worst case is that something will have its height hint (already confirmed, but not yet dropped off from the notifier) updated to a later height.

Reply:
I don't think it'd be a good idea to do so. If we update it for something that already confirmed and attempt to query for its height hint later on, then it'll be off by some margin of blocks, causing us to miss the event.